test/balancer_test.go

```diff
@@ -601,6 +598,7 @@ func (s) TestMetadataInAddressAttributes(t *testing.T) {
 	}
 	defer ss.Stop()
 
+	r := manual.NewBuilderWithScheme("whatever")
```

Comment:
Did you know?  You can inject this into the stubserver and `Dial` will use it:

https://github.com/grpc/grpc-go/blob/230166b6df4ad811bddc77ff5f54251448236ff9/test/end2end_test.go#L5197-L5200

Maybe we should delete that feature since it seems to be well hidden and nonobvious?  Or you should use it here.

Reply:
I knew it. But you cannot set balancer/init service config.
I didn't want to add all the fields just to pass the dial options.

I'm OK with removing it.

Reply:
I'm not seeing where your test needs to set balancer/init service config, though.  You are using a default service config, and then doing the same resolver update that stubServer would do automatically.

Reply:
I'm dumb. I need to set the balancer in init service config, to use the balancer that sets the attributes. But `ss.Start` works. Fixed now.